compilers/concrete-compiler/compiler/lib/Conversion/FHETensorOpsToLinalg/TensorOpsToLinalg.cpp

```diff
@@ -45,11 +45,16 @@ inline void forwardOptimizerID(mlir::Operation *source,
   destination->setAttr("TFHE.OId", optimizerIdAttr);
 }
 
-struct DotToLinalgGeneric
-    : public ::mlir::OpRewritePattern<mlir::concretelang::FHELinalg::Dot> {
-  DotToLinalgGeneric(::mlir::MLIRContext *context)
-      : ::mlir::OpRewritePattern<::mlir::concretelang::FHELinalg::Dot>(
-            context, mlir::concretelang::DEFAULT_PATTERN_BENEFIT) {}
+template <typename DotOp, typename FHEMulOp>
+struct DotToLinalgGeneric : public ::mlir::OpRewritePattern<DotOp> {
```

Comment:
refactored the type of the multiplication op to be able to use this function for both `enc x clear` and `enc x enc`